src/main/java/net/fabricmc/loader/impl/launch/knot/KnotClassDelegate.java

```diff
@@ -71,16 +72,25 @@
 	}
 
 	public void initializeTransformers() {
-		if (transformInitialized) {
-			throw new RuntimeException("Cannot initialize KnotClassDelegate twice!");
+		if (transformInitialized) throw new IllegalStateException("Cannot initialize KnotClassDelegate twice!");
+
+		mixinTransformer = MixinServiceKnot.getTransformer();
+
+		if (mixinTransformer == null) {
+			try { // reflective instantiation for older mixin versions
+				@SuppressWarnings("unchecked")
+				Constructor<IMixinTransformer> ctor = (Constructor<IMixinTransformer>) Class.forName("org.spongepowered.asm.mixin.transformer.MixinTransformer").getConstructor();
+				ctor.setAccessible(true);
+				mixinTransformer = ctor.newInstance();
+			} catch (ReflectiveOperationException e) {
+				throw new IllegalStateException("mixin transformer unavailable?");
```

Comment:
Might as well pass the original exception along too

Reply:
the exception is meant for the original fault (regular mixin interop failing), exceptions from this legacy support hack aren't meaningful outside Mixin 0.8-0.8.2

Reply:
I mean if it does throw a `ReflectiveOperationException` and you're not using Mixin 0.8.3+ presumably you'd want to know why rather than just getting a `IllegalStateException` without a cause

Reply:
There is no better cause really, i updated the commit with a few more comments.